File: app-java/src/main/java/com/open/iconics/application/BaseApplication.java
package com.open.iconics.application;

import android.app.Application;
import android.content.Context;

import com.open.iconics.Iconics;
import com.open.iconics.iconfont.ChangbaIconFont;

public class BaseApplication extends Application {

    private static BaseApplication instance = null;

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        instance = this;
    }

    public static BaseApplication getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        // register custom fonts like this (or also provide a font definition file)
        Iconics.registerFont(new ChangbaIconFont());
    }
}

File: app-kotlin/src/main/java/com/open/iconics/application/BaseApplication.kt
package com.open.iconics.application

import android.app.Application
import android.content.Context
import com.open.iconics.Iconics
import com.open.iconics.iconfont.ChangbaIconFont

class BaseApplication : Application() {

    companion object {

        private lateinit var instance: BaseApplication

        fun getInstance(): BaseApplication = instance
    }

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        instance = this
    }

    override fun onCreate() {
        super.onCreate()
        // DynamicColors.applyToActivitiesIfAvailable(this)

        // register custom fonts like this (or also provide a font definition file)
        Iconics.registerFont(ChangbaIconFont())

        // Generic font creation process
        // GenericFont("GenericFont", "SampleGenericFont", "gnf", "font/materialdrawerfont.ttf")
        //     .also {
        //         it.registerIcon("person", '\ue800')
        //         it.registerIcon("up", '\ue801')
        //         it.registerIcon("down", '\ue802')
        //         Iconics.registerFont(it)
        //    }
    }
}
